posix: fix chmod error on symlinks

xhernandez opened the pull request:
After glibc 2.32 lchmod() is returning EOPNOTSUPP instead of ENOSYS when
called on symlinks. The man page says that the returned code is ENOTSUP.
They are the same in linux, but this patch correctly handles all errors.

Fixes: #2154
Change-Id: Ib3bb3d86d421cba3d7ec8d66b6beb131ef6e0925
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez committed:
posix: fix chmod error on symlinks

After glibc 2.32 lchmod() is returning ENOTSUPP instead of ENOSYS when
called on symlinks. This patch correctly handles this new error.

Fixes: #2154
Change-Id: Ib3bb3d86d421cba3d7ec8d66b6beb131ef6e0925
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez commented:
/run regression

pranithk commented:
/recheck smoke

pranithk commented on xlators/storage/posix/src/posix-inode-fd-ops.c:
```diff
@@ -251,8 +251,13 @@ posix_do_chmod(xlator_t *this, const char *path, struct iatt *stbuf)
+     * is using fchmodat() system call to implement it. However, linux
+     * doesn't support setting the mode for symlinks, so the system
+     * call returns ENOTSUPP. We need to handle both cases. */
+    if ((ret < 0) && ((errno == ENOSYS) || (errno == ENOTSUPP))) {
```
This should be ENOTSUP

xhernandez committed:
Fixed the checked errors

Change-Id: Ic0f182bbdfdae9995e8503eead990fb971cf2070
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez commented:
/run regression

xhernandez commented:
I've updated the first comment to correct the error names. I don't do that on the commit itself to avoid a forced push and a new regression run.

Whoever merges this patch, please copy the commit message from the first comment.

pranithk commented:
> I've updated the first comment to correct the error names. I don't do that on the commit itself to avoid a forced push and a new regression run.
> 
> Whoever merges this patch, please copy the commit message from the first comment.

Default merge is based on squash commit. So it will take whatever is present in this PR description. So I don't think we need to take any special care. Right?

xhernandez commented:
> > I've updated the first comment to correct the error names. I don't do that on the commit itself to avoid a forced push and a new regression run.
> > Whoever merges this patch, please copy the commit message from the first comment.
> 
> Default merge is based on squash commit. So it will take whatever is present in this PR description. So I don't think we need to take any special care. Right?

A Squash merge combines the message from all commits, but I've modified the first comment, not the commit message.

pranithk commented:
@xhernandez Since approvals are already in, anyone can merge. Why don't you do the honors.
